botocore/__init__.py

```diff
@@ -31,11 +31,12 @@ def emit(self, record):
 _first_cap_regex = re.compile('(.)([A-Z][a-z]+)')
 _number_cap_regex = re.compile('([a-z])([0-9]+)')
 _end_cap_regex = re.compile('([a-z0-9])([A-Z])')
+# The regex below handles the special case where some acryonym
+# name is pluralized, e.g GatewayARNs, ListWebACLs, SomeCNAMEs.
+_special_case_transform = re.compile('[A-Z][A-Z][A-Z]+s$')
```

Comment:
Is this basically "match strings that end in 3 or more uppercase characters followed by 's'"? If so, I think this expression is more clear and possibly more efficient: `[A-Z]{2,}s$`

```
>>> re.compile('[A-Z]{2,}s$', re.DEBUG)
max_repeat 2 4294967295
  in
    range (65, 90)
literal 115
at at_end
```

Compared to

```
>>> re.compile('[A-Z][A-Z][A-Z]+s$', re.DEBUG)
in
  range (65, 90)
in
  range (65, 90)
max_repeat 1 4294967295
  in
    range (65, 90)
literal 115
at at_end
```

I haven't benchmarked this to see if it makes a difference, but I think using `[A-Z]{2,}s$` is easier to read.